### PR TITLE
2. Migration - Create experiment API changes for LM and RM.

### PR DESCRIPTION
**Pull request by msvinaykumar**
## Description

create separate tables  for LM and RM.

LM :  kruize_lm_experiments
RM:   kruize_experiments
             
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

**Comment by dinogun** [comment]
@msvinaykumar Please fix the test failures. Looks like some tests need to be updated and some new ones need to be added

**Comment by msvinaykumar** [comment]
> @msvinaykumar Please fix the test failures. Looks like some tests need to be updated and some new ones need to be added

Sure, most of them are expected. I was trying to limit changes in a single PR, but these will be addressed in upcoming PRs. Additionally, some new tests are required, and I’ll add them accordingly.

File: design/KruizeConfiguration.md [hunk 1]
```diff
@@ -139,3 +139,7 @@ The following environment variables are set using the `kubectl apply` command wi
     - Value: "true"
     - Details: This flag is added for getting the details of the inputs passed to the APIs and the corresponding response 
       generated by it. This helps us in debugging the API easily in case of failures.
+- **isROSEnabled**
+  - Description:  if set to True, the ROS application will use the old architecture, where ROS handles experiment creation, result updates, and recommendation updates. If set to False, ROS will utilize the Bulk API, allowing Kruize to manage experiment creation and generate recommendations on its behalf.
```
Comment (chandrams) [inline design/KruizeConfiguration.md:143]: @msvinaykumar - Please update all the remote monitoring tests to enable this. Also add new tests to test RM & LM.

File: src/main/java/com/autotune/database/table/lm/KruizeLMExperimentEntry.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+import com.autotune.analyzer.utils.AnalyzerConstants;
+import com.autotune.database.helper.GenerateExperimentID;
+import com.fasterxml.jackson.databind.JsonNode;
+import jakarta.persistence.*;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+
+/**
+ * This is a Java class named KruizeExperimentEntry annotated with JPA annotations.
```
Comment (chandrams) [inline src/main/java/com/autotune/database/table/lm/KruizeLMExperimentEntry.java:26]: Update the java class, table name & fields

**Comment by dinogun** [comment]
> > @msvinaykumar Please fix the test failures. Looks like some tests need to be updated and some new ones need to be added
> 
> Sure, most of them are expected. I was trying to limit changes in a single PR, but these will be addressed in upcoming PRs. Additionally, some new tests are required, and I’ll add them accordingly.

@msvinaykumar We cant merge PRs which cause PR check failures as it will cause all subsequent PRs to fail as well. Test failures have to be addressed in this PR.  New tests can be added in subsequent PRs

File: tests/scripts/common/common_functions.sh [hunk 3]
```diff
@@ -1919,6 +1919,8 @@ function kruize_remote_patch() {
     sed -i 's/"local": "true"/"local": "false"/' ${KRUIZE_CRC_DEPLOY_MANIFEST_OPENSHIFT}
     sed -i 's/\([[:space:]]*\)\(storage:\)[[:space:]]*[0-9]\+Mi/\1\2 1Gi/' ${KRUIZE_CRC_DEPLOY_MANIFEST_OPENSHIFT}
     sed -i 's/\([[:space:]]*\)\(memory:\)[[:space:]]*".*"/\1\2 "2Gi"/; s/\([[:space:]]*\)\(cpu:\)[[:space:]]*".*"/\1\2 "2"/' ${KRUIZE_CRC_DEPLOY_MANIFEST_OPENSHIFT}
+    sed -i '/"local": "false"/a \ \ \ \ "isROSEnabled": "true",' ${KRUIZE_CRC_DEPLOY_MANIFEST_MINIKUBE}
```
Comment (chandrams) [inline tests/scripts/common/common_functions.sh:1922]: This should be under cluster_type == minikube

File: design/KruizeConfiguration.md (hunk 1 above)
Comment (msvinaykumar) [reply design/KruizeConfiguration.md:143]: done

File: src/main/java/com/autotune/database/table/lm/KruizeLMExperimentEntry.java (hunk 2 above)
Comment (msvinaykumar) [reply src/main/java/com/autotune/database/table/lm/KruizeLMExperimentEntry.java]: done

File: tests/scripts/common/common_functions.sh (hunk 3 above)
Comment (msvinaykumar) [reply tests/scripts/common/common_functions.sh]: done

**Comment by msvinaykumar** [comment]
Duplicate of #1413

**Comment by msvinaykumar** [comment]
Duplicate of https://github.com/kruize/autotune/pull/1413

**Pull request closed without merging**